tests/test_denseadv.cpp

```diff
@@ -0,0 +1,662 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
```

Comment:
almost the same thing for this file: with some small modifications, the standard `test_densad.cpp` should be usable.

(in the context of OPM/ewoms#318, I've heard that additional complexity is considered to be a bad thing. your current solution constitutes a lot of additional complexity and also unnecessarly duplicated code.)